cpp/include/cudf/aggregation.hpp

```diff
@@ -488,5 +489,15 @@ std::unique_ptr<Base> make_merge_sets_aggregation(null_equality nulls_equal = nu
 template <typename Base = aggregation>
 std::unique_ptr<Base> make_merge_m2_aggregation();
 
+/**
+ * @brief Factory to create a CORR aggregation
+ *
+ * Compute correlation matrix amond the input columns.
+ * The input columns are child columns of a non-nullable struct columns.
+ *
+ */
+template <typename Base = aggregation>
+std::unique_ptr<Base> make_corr_aggregation();
```

Comment:
Let's spell out `correlation` instead of `corr`. 
```suggestion
std::unique_ptr<Base> make_correlation_aggregation();
```